docs/markdown/i18n-module.md

```diff
@@ -29,3 +29,58 @@ This merges translations into a text file using `msgfmt`. See [custom_target](ht
 * `type`: type of file, valid options are `'xml'` (default) and `'desktop'`
 
 *Added 0.37.0*
+
+### i18n.create_pot()
+
+```meson
+  i18n.create_pot(name)
+```
+
+*Added 0.42.0*
+
+These are all the supported keyword arguments:
+
+* `languages`: list of languages that are to be generated. As of 0.37.0 this is optional and the [LINGUAS](https://www.gnu.org/software/gettext/manual/html_node/po_002fLINGUAS.html) file is read.
+* `data_dirs`: list of directories to be set for `GETTEXTDATADIRS` env var (Requires gettext 0.19.8+), used for local its files
+* `preset`: name of a preset list of keywords, flags and arguments, current option is `'glib'`, see [source](https://github.com/mesonbuild/meson/blob/master/mesonbuild/modules/i18n.py) for their value
+
+Returns a pot file object, that you must fill with sets of files. This object has the following methods:
+
+#### pot.add_files()
+
+```meson
+  pot.add_files(*file1*, *file2*, ...)
+```
+
+This method adds a set of files to the pot file. This means these files will be seached for string using `xgettext`.
+These are all the supported keyword arguments:
+
+* `language`: the (computer) language of the files
+* `keywords`: list of keywords used to extract strings (use the special `''` value to disable default keywords)
+* `flags`: list of flags used to extract strings
+* `extra_args`: list of extra arguments to pass to `xgettext`
+* `preset`: name of a preset list of keywords, flags and arguments (see `i18n.create_pot()`)
+
+#### pot.merge_files()
+
+```meson
+  pot.merge_files(*file1*, *file2*, ...)
+```
+
+This method adds targets to generate files with translations using `msgfmt`. The translations are searched from the pot object directory.
+These are all the supported keyword arguments:
+
+* `suffix`: this suffix will be stripped from file names to get the output file name and target name
+* `language`: the (computer) language of the files, valid options are `'xml'` (default) and `'desktop'`
+* `keywords`: list of keywords used to merge strings (use the special `''` value to disable default keywords)
+* `extra_args`: list of extra arguments to pass to `msgfmt`
+* `preset`: name of a preset list of keywords, flags and arguments (see `i18n.create_pot()`)
+
+#### pot.add_merge_files()
+
+```meson
+  pot.add_merge_files(*file1*, *file2*, ...)
+```
+
+This method is a shortcut to `pot.add_files()` then `pot.merge_files()` on a set of files.
```

Comment:
Not me, but there is `i18n.merge_file()` and you’re not forced to call `i18n.gettext()` so I just wanted to keep that option.
Also, you cannot use `extra_args` iwth `pot.add_merge_files()` because there is no way to know which are for the `xgettext` call and which are for the `msgfmt` call. Since I wanted to keep `extra_args` as the standard name, and not having `xgettext_extra_args` and `msgfmt_extra_args`, this is the simplest way.

Just let me know what you prefer I’ll happily rework the code to fit. :-)